### PR TITLE
Instance stats service composition root

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Ability to create a dependency graph of the instance stats service in one method call. This PR shows high coupling to 10+ stores we have in this part of the code.

### CODE REVIEW

File: src/test/e2e/api/admin/instance-admin.e2e.test.ts [hunk 1]
```diff
@@ -20,7 +33,7 @@ afterAll(async () => {
 });
 
 test('should return instance statistics', async () => {
-    stores.featureToggleStore.create('default', {
+    await stores.featureToggleStore.create('default', {
```
Comment (kwasniew) [inline src/test/e2e/api/admin/instance-admin.e2e.test.ts:36]: missing await

File: src/test/e2e/api/admin/instance-admin.e2e.test.ts [hunk 2]
```diff
@@ -11,7 +14,17 @@ beforeAll(async () => {
     db = await dbInit('instance_admin_api_serial', getLogger);
     stores = db.stores;
     await stores.settingStore.insert('instanceInfo', { id: 'test-static' });
-    app = await setupApp(stores);
+    app = await setupAppWithCustomConfig(
+        stores,
+        {
+            experimental: {
+                flags: {
+                    strictSchemaValidation: true,
```
Comment (kwasniew) [inline src/test/e2e/api/admin/instance-admin.e2e.test.ts:22]: missing schema validation

File: src/test/e2e/helpers/database-init.ts [hunk 3]
```diff
@@ -97,6 +98,7 @@ export default async function init(
         getLogger,
     });
 
+    log.setLogLevel('error');
```
Comment (kwasniew) [inline src/test/e2e/helpers/database-init.ts:101]: stop logging db migration messages in all tests

**Review by daveleek — APPROVED** [review]
Nice one! Yeah that's a lot of dependencies 😅

File: src/test/e2e/helpers/database-init.ts (hunk 3 above)
Comment (gastonfournier) [reply src/test/e2e/helpers/database-init.ts:101]: Awesome! This was something I was planning to do yesterday :1st_place_medal:

File: src/lib/services/index.ts [hunk 4]
```diff
@@ -299,13 +303,9 @@ export const createServices = (
...
-    const instanceStatsService = new InstanceStatsService(
-        stores,
-        config,
-        versionService,
-        db ? createGetActiveUsers(db) : createFakeGetActiveUsers(),
-        db ? createGetProductionChanges(db) : createFakeGetProductionChanges(),
-    );
+    const instanceStatsService = db
+        ? createInstanceStatsService(db, config)
+        : createFakeInstanceStatsService(config);
```
Comment (gastonfournier) [inline src/lib/services/index.ts:308]: Neat!

**Review by gastonfournier — APPROVED** [review]

**Pull request merged**